integrations/huggingface-transformers/tutorials/sparse-transfer-learning-bert.md

```diff
@@ -0,0 +1,343 @@
+
+## Installation
+
+Install via `pip`:
+
+```
+pip install sparseml[torch]
+```
+
+## Sparse Transfer Learning onto SST2
+
+Let's try a simple example of fine-tuning a pre-sparsified model onto the SST dataset. SST2 is a sentiment analysis
+dataset, with pairs of sentences and binary labels representing positive or negative sentiment.
+
+[SST2 Dataset Card](https://huggingface.co/datasets/glue/viewer/sst2/train)
+
+### Selecting a Pre-sparsified Upstream Model
+
+We will fine-tune a [90% sparsified BERT-base](https://arxiv.org/abs/2203.07259). This model is hosted in [SparseZoo](https://sparsezoo.neuralmagic.com/models/nlp%2Fmasked_language_modeling%2Fobert-base%2Fpytorch%2Fhuggingface%2Fwikipedia_bookcorpus%2Fpruned90-none) and is identified by the following stub:
+```bash
+zoo:nlp/masked_language_modeling/obert-base/pytorch/huggingface/wikipedia_bookcorpus/pruned90-none
+```
+
+SparseML uses the stub to download the model before starting training.
+
+### Kick off Training
+
+Run the following to fine-tune the pre-sparsified BERT onto SST2:
+
+```bash
+sparseml.transformers.text_classification \
+  --output_dir pruned_quantized_obert-text_classification_sst2 \
+  --model_name_or_path zoo:nlp/masked_language_modeling/obert-base/pytorch/huggingface/wikipedia_bookcorpus/pruned90-none \
+  --recipe zoo:nlp/sentiment_analysis/obert-base/pytorch/huggingface/sst2/pruned90_quant-none \
+  --distill_teacher zoo:nlp/sentiment_analysis/obert-base/pytorch/huggingface/sst2/base-none \
+  --task_name sst2 --max_seq_length 128 --per_device_train_batch_size 32 --per_device_eval_batch_size 32 --preprocessing_num_workers 6 \
+  --do_train --do_eval --evaluation_strategy epoch --fp16  \
+  --save_strategy epoch --save_total_limit 1
+```
+
+Lets discuss the key arguments:
+
+- `--task` specifies the dataset to train on (in this case SST2). You can pass any GLUE task to the `--task` command. Check out the use case pages for details on passing a custom dataset.
+
+- `--model_name_or_path` specifies the starting checkpoint for the training process. Here, we passed a SparseZoo stub, which
+identifies the 90% pruned BERT model in the SparseZoo. The script downloads the PyTorch model to begin training. In addition to SparseZoo stubs, you can also pass a local path to a PyTorch checkpoint.
+
+- `--recipe` specifies the transfer learning recipe. Recipes are YAML files that declare the sparsity related algorithms
+ that SparseML should apply. For transfer learning, the recipe instructs SparseML to maintain sparsity during training
+ and to apply quantization over the final epochs. In this case, we passed a SparseZoo stub, which instructs SparseML
+ to download a premade SST2 transfer learning recipe. In addition to SparseZoo stubs, you can also pass a local path to a YAML recipe. 
```

Comment:
In this case, we passed a SparseZoo stub instructing SparseML to download a premade SST2 transfer learning recipe.